server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java

```diff
@@ -207,7 +207,11 @@ void checkLimits(IndexSettings settings) {
     }
 
     private void checkFieldLimit(long limit) {
-        if (fieldMappers.size() + objectMappers.size() - mapping.getSortedMetadataMappers().length > limit) {
+        checkFieldLimit(limit, 0);
+    }
+
+    public void checkFieldLimit(long limit, int additionalFieldsToAdd) {
```

Comment:
nit: I find it a bit odd that we take an int but the args are always 0 or 1. maybe we should have two methods "checkFieldLimit" and "canAddNewField" or something along those lines? Also I would prefer that this method is package private. MappingLookup is not really meant to expose these to the outside world (at least outside its package).

Reply:
>but the args are always 0 or 1.

The ParseContext class passes `dynamicMappers.size() + 1` not 1.

Reply:
I've made the method package-private (see 8566c61). As Mark said, we're not just passing 0 and 1.

Reply:
right, my eyes tricked me there :)